piptools/utils.py

```diff
@@ -695,11 +686,27 @@ def parse_config_file(config_file: Path) -> dict[str, Any]:
             hint=f"Could not parse '{config_file !s}': {value_err !s}",
         )
 
+    cli_opts = {
+        opt: option
+        for option in click_context.command.params
+        for opt in itertools.chain(option.opts, option.secondary_opts)
+    }
+
     # In a TOML file, we expect the config to be under `[tool.pip-tools]`
     piptools_config: dict[str, Any] = config.get("tool", {}).get("pip-tools", {})
-    piptools_config = {
-        get_click_dest_for_option(k): v for k, v in piptools_config.items()
-    }
+
+    # Replace boolean flags like ``--no-annotate`` with their equivalents
+    try:
```

Comment:
if we run `_validate_config()` function before this line then we won't need the try-except block.

Reply:
This isn't possible since the boolean flags aren't filtered at this point...

Reply:
Yeah, this would require to update `_validate_config()`.

Reply:
I've checked and it raises error on `no-index`:

```diff
diff --git piptools/utils.py piptools/utils.py
index f454ce7..470f844 100644
--- piptools/utils.py
+++ piptools/utils.py
@@ -701,6 +701,8 @@ def parse_config_file(
     # In a TOML file, we expect the config to be under `[tool.pip-tools]`
     piptools_config: dict[str, Any] = config.get("tool", {}).get("pip-tools", {})

+    _validate_config(click_context, piptools_config)
+
     # Replace boolean flags like ``--no-annotate`` with their equivalents
     try:
         piptools_config = {
```

```bash
$ cat .pip-tools.toml
[tool.pip-tools]
no-index = true

$ pip-compile
Usage: pip-compile [OPTIONS] [SRC_FILES]...
Try 'pip-compile -h' for help.

Error: No such config key 'no-index'. Did you mean no_index?
```

Reply:
Oh, wait `no-index=True` should be valid flag.